src/components/FormDialog.tsx

```diff
@@ -256,8 +256,10 @@ const FormDialog = <T extends Record<string, unknown>>({
           if (meta.conditional && !data[meta.conditional]) {
             return <div key={name} />;
           }
-          const setValue = (value: unknown) =>
-            setData((d) => ({ ...d, [name]: value }));
+          const setValue = useCallback(
+            (value: unknown) => setData((d) => ({ ...d, [name]: value })),
+            [setData]
+          );
```

Comment:
I don't think this works bc hooks could only be called on the top level - you can verify by running `npm run start` and loading `roamjs-components` like any extension, then run the `Render Form Dialog` command from the Roam Command Palette

Reply:
Seems to work! 👍